### PR TITLE
Update to AsyncApi 3.0

### LINKED ISSUE #494: Update to AsyncApi spec 3.0
**Describe the feature request**
Update to AsyncApi spec 3.0

**Motivation**
The spec got released and we want to update to the latest version

**Status**
- [x] Update backend AsyncApi pojos
- [x] Update springwolf-ui

**Technical details**
- https://www.asyncapi.com/blog/async-api-spec-3.0-release
- https://www.asyncapi.com/docs/migration/migrating-to-v3

**Describe alternatives you've considered**
None

### PR DESCRIPTION
**Pull request by timonback**
This PR + feature branch is intended to allow the migration to AsyncApi 3.0 step-by-step.

Feel free to open any PR against this feature branch (instead of the master branch).

Preview: https://deploy-preview-507--springwolf-ui.netlify.app/springwolf-ui/asyncapi-ui.html

Verification: https://studio.asyncapi.com using i.e. `https://raw.githubusercontent.com/springwolf/springwolf-core/feat/asyncapi-3/springwolf-examples/springwolf-kafka-example/src/test/resources/asyncapi.json`

Closes #494

### CODE REVIEW

**Comment by timonback** [comment]
Collection of items missing/worth for discussion:
- [ ] Do we place `messages` inside of the components (next to schema) vs keeping it in the channel (duplicating them in the channel is likely easier) -> keep it in the channel
- [ ] All operations need an action
- [ ] All operations need a channel ref
- [ ] All operation keys require at least topic name, action, method (operation identifier) - since by convention all messages[] arrays contain exactly one entry
- [ ] AsyncListener metadata is basically a OperationBinding -> ui will read metadata from the operation (i.e. description)

File: springwolf-add-ons/springwolf-common-model-converters/src/main/java/io/github/stavshamir/springwolf/addons/common_model_converters/converters/monetaryamount/MonetaryAmount.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class MonetaryAmount {
 
     @JsonProperty("amount")
-    @Schema(example = "99.99")
+    @Schema(example = "99.99", minimum = "0.1")
```
Comment (ctasada) [inline springwolf-add-ons/springwolf-common-model-converters/src/main/java/io/github/stavshamir/springwolf/addons/common_model_converters/converters/monetaryamount/MonetaryAmount.java:13]: Any specific reason the minimum cannot be "0.0"?

File: springwolf-asyncapi/src/main/java/io/github/stavshamir/springwolf/asyncapi/v3/model/channel/message/MessageObject.java [hunk 2]
```diff
@@ -118,4 +118,14 @@ public class MessageObject extends ExtendableObject implements Message {
      */
     @JsonProperty(value = "traits")
     private List<MessageTrait> traits;
+
+    /*
+     * Override the getMessageId to guarantee that there's always a value. Defaults to 'name'
+     */
+    public String getMessageId() {
```
Comment (ctasada) [inline springwolf-asyncapi/src/main/java/io/github/stavshamir/springwolf/asyncapi/v3/model/channel/message/MessageObject.java:125]: The `messageId` is **not** part of the `MessageObject` definition. The AsyncAPI v3 defines `messageId` as a key when a `Message` is part of a `Map`.

The `messageId` field was added to automatically build the maps when needed, but this method was needed to fix some cases.

Should we take a similar approach as the one with `ChannelId` and also remove this value and _always_ use the `name` field?

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java [hunk 3]
```diff
@@ -10,5 +11,10 @@ public interface ChannelsScanner {
     /**
      * @return A mapping of channel names to their respective channel object for a given protocol.
      */
-    Map<String, ChannelItem> scan();
+    Map<String, ChannelObject> scanChannels();
+
+    /**
+     * @return A mapping of operation names to their respective operation object for a given protocol.
+     */
+    Map<String, Operation> scanOperations();
```
Comment (ctasada) [inline springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java:19]: Now the `ChannelScanner` scans for both `channels` and `operations`. Those components are decoupled in the new AsyncAPI v3 spec, so maybe would make sense to have a standalone `OperationsScanner`.

I think would be good to define it before the 1.0 release.

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/AsyncMessage.java [hunk 4]
```diff
@@ -1,45 +1,43 @@
...
+     * Mapped to {@link MessageObject#getName()}
      */
     String name() default "";
 
     /**
-     * Mapped to {@link Message#getSchemaFormat()}
+     * Mapped to {@link MessageObject#getContentType()}
      */
-    String schemaFormat() default DEFAULT_SCHEMA_FORMAT;
+    String contentType() default "application/json";
```
Comment (ctasada) [inline springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/AsyncMessage.java:37]: This is a breaking change. Should be documented.

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/types/AsyncAPI.java [hunk 5]
```diff
@@ -71,25 +74,30 @@ public class AsyncAPI {
...
+     */
+    @JsonProperty(value = "operations")
+    private Map<String, Operation> operations;
 
     /**
      * Holds a set of reusable objects for different aspects of the AsyncAPI specification.
      * All objects defined within the components object will have no effect on the API unless they are explicitly
      * referenced from properties outside the components object.
      */
     private Components components;
```
Comment (ctasada) [inline springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/types/AsyncAPI.java:102]: I didn't manage to remove this AsyncAPI class and replace it by the one from the `springwolf-asyncapi` module. The reason is that this `components` field is used in different places. 

Maybe would be worth it to drop a comment here if this class is intended to stay _"duplicated"_

File: springwolf-ui/README.md [hunk 6]
```diff
@@ -16,6 +16,13 @@ dependencies {
 
 After starting the application, visit: `localhost:8080/springwolf/asyncapi-ui.html`.
 
+## TODOs:
+
+- Migrate to AsyncApi 3
```
Comment (ctasada) [inline springwolf-ui/README.md:21]: Is this TODO still valid?

File: springwolf-plugins/springwolf-sqs-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SqsAsyncQueueBinding.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER})
+@Inherited
+public @interface SqsAsyncQueueBinding {
```
Comment (ctasada) [inline springwolf-plugins/springwolf-sqs-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SqsAsyncQueueBinding.java:13]: Requires to be added to the documentation

File: springwolf-plugins/springwolf-sqs-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SqsAsyncOperationBinding.java [hunk 8]
```diff
@@ -20,4 +20,6 @@
 public @interface SqsAsyncOperationBinding {
 
     String type() default "sqs";
+
+    SqsAsyncQueueBinding[] queues() default {};
```
Comment (ctasada) [inline springwolf-plugins/springwolf-sqs-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SqsAsyncOperationBinding.java:24]: Requires to be added to the documentation

File: springwolf-plugins/springwolf-sns-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SnsAsyncOperationBinding.java [hunk 9]
```diff
@@ -20,4 +20,10 @@
 public @interface SnsAsyncOperationBinding {
 
     String type() default "sns";
+
+    String protocol();
+
+    SnsAsyncOperationBindingIdentifier endpoint();
```
Comment (ctasada) [inline springwolf-plugins/springwolf-sns-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/operationdata/annotation/SnsAsyncOperationBinding.java:26]: Both `protocol` and `endpoint` are breaking changes. Documentation needs to be updated.

File: springwolf-add-ons/springwolf-common-model-converters/src/main/java/io/github/stavshamir/springwolf/addons/common_model_converters/converters/monetaryamount/MonetaryAmount.java (hunk 1 above)
Comment (timonback) [reply springwolf-add-ons/springwolf-common-model-converters/src/main/java/io/github/stavshamir/springwolf/addons/common_model_converters/converters/monetaryamount/MonetaryAmount.java]: No reason, this demonstrates an new field in springwolf-ui
Changing this should be easy

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java (hunk 3 above)
Comment (timonback) [reply springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java:19]: Good question. I believe they most likely come in pairs, still I am in favour of splitting this into two interfaces.

File: springwolf-ui/README.md (hunk 6 above)
Comment (timonback) [reply springwolf-ui/README.md:21]: The ui side is done

File: springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java (hunk 3 above)
Comment (ctasada) [reply springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java:19]: I'm preparing a PR with this change. It's a bit more work than expected 😄

Comment (ctasada) [reply springwolf-core/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/ChannelsScanner.java:19]: @timonback please take a look to https://github.com/springwolf/springwolf-core/pull/562 it became larger than expected

**Pull request merged**